jan2025/helpers.js

```diff
@@ -0,0 +1,369 @@
+// Follows details outlined in gsheet here https://docs.google.com/spreadsheets/d/1Vtxp5BLweDPDooQoNhgOCYjCIBPRYIcOuyArGJRqOkI/edit?gid=0#gid=0
+const migrateFlowData = (flowData) => {
+  const timestamp = `[${new Date().toISOString().replace(/T/, ' ').replace(/\..+/, '')}]`;
+
+  let newFlowData = flowData;
+  let logs = "";
+
+  Object.entries(flowData).forEach(([nodeId, nodeData]) => {
```

Comment:
If a single one of these steps throw an error (they _shouldn't_ due to good optional chaining on your part below 👍 ), the entire function will fail.

It might be worth wrapping each step in a try/catch just for debugging - if there's a point of failure, we can more easily find it.

```js
try {
  // logic here
} catch (error) {
  logs += `${timestamp} Failed to update node ${nodeId}). Error: ${error}; `;
  throw error;
}
```